infoblox/resource_infoblox_zone_auth.go

```diff
@@ -0,0 +1,179 @@
+				Optional:    true,
+				Description: "Time in seconds for secondary servers to cache data for 'Does Not Respond' responses",
+			},
+			"soa_refresh": {
+				Type:        schema.TypeInt,
+				Default:     10800,
+				Optional:    true,
+				Description: "Interval in seconds for secondary servers to check the primary server for fresh data about the zone",
+			},
+			"soa_retry": {
+				Type:        schema.TypeInt,
+				Default:     3600,
+				Optional:    true,
+				Description: "Interval in seconds for secondary servers to wait before recontacting primary server about the zone after failure",
+			},
+			"ext_attrs": {
+				Type:        schema.TypeString,
+				Default:     "",
+				Optional:    true,
+				Description: "Extensible attributes, as a map in JSON format",
+			},
+		},
+	}
+}
+
+// CRUD FUNCTIONS
+
+func resourceZoneAuthCreate(d *schema.ResourceData, m interface{}) error {
+	fqdn := d.Get("fqdn").(string)
+	nsGroup := d.Get("ns_group").(string)
+	restartIfNeeded := d.Get("restart_if_needed").(bool)
+	comment := d.Get("comment").(string)
+	soaDefaultTtl := d.Get("soa_default_ttl").(int)
+	soaExpire := d.Get("soa_expire").(int)
+	soaNegativeTtl := d.Get("soa_negative_ttl").(int)
+	soaRefresh := d.Get("soa_refresh").(int)
+	soaRetry := d.Get("soa_retry").(int)
+
+	extAttrJSON := d.Get("ext_attrs").(string)
+	extAttrs := make(map[string]interface{})
+	if extAttrJSON != "" {
+		if err := json.Unmarshal([]byte(extAttrJSON), &extAttrs); err != nil {
+			return fmt.Errorf("cannot process 'ext_attrs' field: %s", err.Error())
+		}
+	}
+
+	var tenantID string
+	tempVal, found := extAttrs["Tenant ID"]
+	if found {
+		tenantID = tempVal.(string)
+	}
+	connector := m.(ibclient.IBConnector)
+	objMgr := ibclient.NewObjectManager(connector, "Terraform", tenantID)
+
+	newZone, err := objMgr.CreateZoneAuth(
+		fqdn, nsGroup, restartIfNeeded, comment, soaDefaultTtl, soaExpire, soaNegativeTtl, soaRefresh, soaRetry, extAttrs)
+	if err != nil {
+		return fmt.Errorf("error creating Zone Auth: %s", err.Error())
+	}
+
+	d.SetId(newZone.Ref)
+
+	return nil
+}
+
+func resourceZoneAuthGet(d *schema.ResourceData, m interface{}) error {
+	extAttrJSON := d.Get("ext_attrs").(string)
+	extAttrs := make(map[string]interface{})
+	if extAttrJSON != "" {
+		if err := json.Unmarshal([]byte(extAttrJSON), &extAttrs); err != nil {
+			return fmt.Errorf("cannot process 'ext_attrs' field: %s", err.Error())
+		}
+	}
+
+	var tenantID string
+	tempVal, found := extAttrs["Tenant ID"]
+	if found {
+		tenantID = tempVal.(string)
+	}
+	connector := m.(ibclient.IBConnector)
+	objMgr := ibclient.NewObjectManager(connector, "Terraform", tenantID)
+
+	obj, err := objMgr.GetZoneAuthByRef(d.Id())
+	if err != nil {
+		return fmt.Errorf("getting Zone Auth failed: %s", err.Error())
+	}
+
+	d.Set("fqdn", obj.Fqdn)
+	d.Set("comment", obj.Comment)
+	d.SetId(obj.Ref)
+
+	return nil
+}
+
+func resourceZoneAuthUpdate(d *schema.ResourceData, m interface{}) error {
+	return nil
```

Comment:
This is a bad way for not supporting updates for a resource. Please, change this, either by writing the actual code for 'UPDATE' operation or by clearly indicating that the resource update is not supported so far.

Reply:
I agree; I've created the Update function in the Go Client and used it here. More details in the comment below.